names/names.py

```diff
@@ -2,6 +2,41 @@
 
 start_time = time.time()
 
+class BinarySearchTree:
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
+
+    # Insert the given value into the tree
+    def insert(self, value):
+        if value >= self.value:
+            if self.right is None:
+                self.right = BinarySearchTree(value)
+            else:
+                self.right.insert(value)
+        else:
+            if self.left is None:
+                self.left = BinarySearchTree(value)
+            else:
+                self.left.insert(value)
+
+    # Find if the value is already in the tree
+    def dupeFind(self, value):
+        if value > self.value:
+            if self.right is None:
+                pass
+            else:
+                self.right.dupeFind(value)
+        elif value < self.value:
+            if self.left is None:
+                pass
+            else:
+                self.left.dupeFind(value)
+        # If the value is in the tree, add it to the duplicates array
+        else:
+            duplicates.append(value)
```

Comment:
I think it would be a better solution to abstract out the logic that is appending duplicate values only because of OOP principles of abstraction. You wouldn't want to mix your logic for finding the duplicates into your BST so that it could be reusable in other situations rather than just finding duplicates. I think that's what tripped me up with how you solved this. Other than the it's the exact same logic as what I sent you except keeping the action of appending duplicates outside of the class.